src/communication/handlers.ts

```diff
@@ -69,6 +70,34 @@ function buildNotification (agreement: Agreement, message: CommsMessage<CommsPay
   }
 }
 
+async function getAgreementWithRetry (
+  agreementRef: string,
+  retriesCount: number,
+  retryInterval: number
+): Promise<Agreement | undefined> {
+  function getAgreement (): Promise<Agreement> {
+    logger.info('Trying to get agreement ' + agreementRef)
+    return Agreement
+      .findOne({ where: { agreementReference: agreementRef } })
+      .then(agreement => {
+        if (!agreement) {
+          throw new Error('Agreement not found')
+        }
+        return agreement
+      })
+  }
+  return await asyncRetry<Agreement>(
+    getAgreement,
+    retriesCount,
+    retryInterval
+  ).catch(e => {
+    if (e.message.indexOf('Max retries reached for function') !== -1) {
+      return undefined
```

Comment:
```suggestion
      return
```

Undefined is not necessary...

Reply:
Done